test/templates/realtime.tpl.go

```diff
@@ -22,11 +22,15 @@ func main() {
 		os.Exit(1)
 	}
 
-	go func() {
-		time.Sleep(1 * time.Second)
-		fmt.Println("Timeout exceeded")
-		os.Exit(2)
-	}()
+	rescueStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
 
 	[snippet-code]
+
+	b := make([]byte, 4096)
+	n, _ := r.Read(b)
```

Comment:
Beware: this is a blocking operation, meaning that the code will stops until killed if a test fails to print anything to the standard output (e.g. if a test prints on the standard error stream)

I don't know Go much, but either there is a non-blocking Read method (or peek or something), or you have to use a goroutine with a proper retry loop